lib/error.sh

```diff
@@ -0,0 +1,217 @@
+# Copyright (c) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Description: Error handling functions.
+
+# Write specified message to stderr.
+stderr()
+{
+	local msg="$*"
+
+	echo >&2 "$msg"
+}
+
+# Simplified version of die that should be called by functions that
+# could fail as part of the normal "die()" code path. Required to
+# avoid infinite recursion.
+_fatal()
+{
+	echo >&2 "FATAL: $msg"
+	exit 1
+}
+
+# Canonicalize the specified path (which must be valid).
+resolve_path()
+{
+	local file="${1:-}"
```

Comment:
Looks to me like this could be simplified as:

```
local path=$(readlink --canonicalize-existing "$1" || _fatal "need file to resolve, got $1")
echo $path
```

* Testing the empty case is not necessary, since readlink itself errors out in that case
* Checking for `/` is not necessary, since `-e` is a shortcut for `--canonicalize-existing`, which itself does canonicalize

I also used the extended option name, usually clearer in scripts.